src/table/table-model.class.ts

```diff
@@ -149,6 +154,64 @@ export class TableModel implements PaginationModel {
 	 */
 	protected _data: TableItem[][] = [[]];
 
+	constructor() {
+		TableModel.COUNT++;
+	}
+
+	/**
+	 * Returns an id for the given column
+	 *
+	 * @param column the column to generate an id for
+	 * @param row the row of the header to generate an id for
+	 */
+	getId(column: number | "select" | "expand", row = 0): string {
```

Comment:
can `column` use enums or interfaces?

Reply:
not sure how I'd apply an interface to this ... Enum would be mostly the same (union with `number | HeaderTypeEnum`) so I'm not sure it's a strict advantage

Reply:
Right now we've got
`getId(column: number | "select" | "expand", row = 0): string {`
and
`getHeaderId(column: number | "select" | "expand", colSpan = 1): string {`
that repeat the inline enum.

We should define somewhere else and re-use

Reply:
There's also not a clean way to apply enum values from a template ...

Reply:
Can't we just do `getHeaderId(column: number | HeaderTypeEnum, colSpan = 1): string {`?

Reply:
Sure, but in each component that calls that method in the template we'd have to keep a local `headerTypeEnum` on the class ... not super ideal

Reply:
We get the same type guarantees with string unions as with enums in this case, so there's not a huge advantage (and it's not like we'll have 50 string options)

Reply:
What we _can_ do is add a `type` definition so we can reuse it

Reply:
so: `type HeaderType = number | "select" | "expand"`

and `getHeaderId(column: HeaderType`